#### 5. Устройство состоит из трех деталей. Для первой детали вероятность выйти из строя в первый месяц равна 0.1, для второй - 0.2, для третьей - 0.25. Какова вероятность того, что в первый месяц выйдут из строя:

-  все детали;  
-  только две детали;  
-  хотя бы одна деталь;
-  от одной до двух деталей

Это событие может произойти следующими способами:  
$P(A_1)$: вышла из строя первая деталь, вероятность этого события: $P(A_1)=0.1$;  
$P(\bar A_1)$: противоположное событию $P(A_1)$, первая деталь не вышла из строя, вероятность этого события: $P(\bar A_1)=1-P(A_1)=1-0.1=0,9$;  

$P(A_2)$: вышла из строя вторая деталь, вероятность этого события: $P(A_2)=0.2$;  
$P(\bar A_2)$: противоположное событию $P(A_2)$, вторая деталь не вышла из строя, вероятность этого события: $P(\bar A_2)=1-P(A_2)=1-0.2=0,8$;  

$P(A_3)$: вышла из строя третья деталь, вероятность этого события: $P(A_3)=0.25$;  
$P(\bar A_3)$: противоположное событию $P(A_3)$, третья деталь не вышла из строя, вероятность этого события: $P(\bar A_3)=1-P(A_3)=1-0.25=0,75$;  

а) Вероятность того, что вышли из строя все детали, произойдет если события $A_1$, $A_2$, и $A_3$ произойдут одновременно: $P(A)=P(A_1)\cdot P(A_2)\cdot P(A_3)=0.1\cdot 0.2\cdot 0.25$

In [9]:
PA = 0.1 * 0.2 * 0.25
print(f'Вероятность того, что из строя выйдут все детали Р(A) = {round(PA, 4)} = {round(PA * 100, 2)}%')

Вероятность того, что из строя выйдут все детали Р(A) = 0.005 = 0.5%


б) Вероятность того, что вышли из строя только две детали, если:  
-  Произошли события $P(A_1)$ $И$ $P(A_2)$ $И$ $P(\bar A_3)$ $ИЛИ$;  
-  Произошли события $P(A_1)$ $И$ $P(A_3)$ $И$ $P(\bar A_2)$ $ИЛИ$;  
-  Произошли события $P(A_2)$ $И$ $P(A_3)$ $И$ $P(\bar A_1)$.  
   
А вероятностью этого события  $P(B)$  будет суммой вероятностей этих комбинаций событий.  
$P(B)=P(A_1)\cdot P(A_2)\cdot P(\bar A_3)+P(A_1)\cdot P(A_3)\cdot P(\bar A_2)+P(A_2)\cdot P(A_3)\cdot P(\bar A_1)=0.1\cdot 0.2\cdot 0.75+0.1\cdot 0.25\cdot 0.8+0.2\cdot 0.25\cdot 0.9$

In [8]:
PB = 0.1 * 0.2 * 0.75 + 0.1 * 0.25 * 0.8 + 0.2 * 0.25 * 0.9
print(f'Вероятность того, что из строя выйдут только 2 детали Р(B) = {round(PB, 4)} = {round(PB * 100, 2)}%')

Вероятность того, что из строя выйдут только 2 детали Р(B) = 0.08 = 8.0%


в) Вероятность того, что вышли из строя  хотя бы одна деталь, как противоположное событию не вышло из строя ни одной детали:  

Вероятность того, что не вышло из строя ни одной детали $\bar A_1$, $\bar A_2$, и $\bar A_3$. Произойдут одновременно: $P(C)=P(\bar A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)=0.9*0.8*0.75$

In [7]:
PC = 0.9 * 0.8 * 0.75
print(f'Вероятность того, что из строя не выйдет ни одной детали Р(C) = {round(PC, 4)} = {round(PC * 100, 2)}%')

Вероятность того, что из строя не выйдет ни одной детали Р(C) = 0.54 = 54.0%


Вероятность того, что вышли из строя хотя бы одна деталь $P(\ge 1) = 1 - P(C)$

In [6]:
PD = 1 - PC
print(f'Вероятность того, что вышли из строя хотя бы одна деталь PD = {round(PD, 4)} = {round(PD * 100, 2)}%')

Вероятность того, что вышли из строя хотя бы одна деталь PD = 0.46 = 46.0%


г) Вероятность того, что вышли из строя выйдут от одной до двух деталей, если: 
-  Произошли события $P(A_1)$ $И$ $P(\bar A_2)$ $И$ $P(\bar A_3)$ $ИЛИ$;  
-  Произошли события $P(A_2)$ $И$ $P(\bar A_1)$ $И$ $P(\bar A_3)$ $ИЛИ$;  
-  Произошли события $P(A_3)$ $И$ $P(\bar A_1)$ $И$ $P(\bar A_2)$.  
   
А вероятностью этого события  $P(E)$  будет **суммой** вероятностей этих комбинаций событий.  
$P(E)=P(A_1)\cdot P(\bar A_2)\cdot P(\bar A_3)+P(A_2)\cdot P(\bar A_1)\cdot P(\bar A_3)+P(A_3)\cdot P(\bar A_1)\cdot P(\bar A_2)=0.1\cdot 0.8\cdot 0.75+0.2\cdot 0.9\cdot 0.75+0.25\cdot 0.9\cdot 0.8$  

$P(F) = P(D) + P(E) = 0.375 + 0.08$

In [12]:
PE = 0.1 * 0.8 * 0.75 + 0.2 * 0.9 * 0.75 + 0.25 * 0.9 * 0.8
PF = PE + PB
print(f'Вероятность того, что выйдет из строя одна деталь Р(E) = {round(PF, 4)} = {round(PF * 100, 2)}%')

Вероятность того, что выйдет из строя одна деталь Р(E) = 0.455 = 45.5%


### Результат:  

а)  Вероятность того что в первый месяц выйдут из строя все детали 0.50%  
б)  Вероятность того что в первый месяц выйдут из строя 2 детали 8.00%  
в)  Вероятность того что в первый месяц выйдут из строя хотя бы одна деталь 46.00%   
г)  Вероятность того что в первый месяц выйдут из строя от одной до двух деталей 45.50%   